runtime/compiler/runtime/JITServerROMClassHash.cpp

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "j9.h"
+#include "net/LoadSSLLibs.hpp"
+#include "runtime/JITServerROMClassHash.hpp"
+
+
+JITServerROMClassHash::JITServerROMClassHash(const J9ROMClass *romClass)
+{
```

Comment:
Doubt it, but I'll see if the context can be used safely in multiple threads. If so, I'll create it in a static init function.

Reply:
Looks like it's not thread safe, so it could only be cached per compilation thread. Based on https://github.com/openssl/openssl/issues/7219, the overhead is significant when the data size is small, but I doubt it makes a difference for us since ROMClasses are a few KBs on average. I don't think the complexity of caching the context per thread is worth it unless we actually measure the overhead to be non-negligible. I will also change the code to use a simpler API `EVP_Digest()` that is supposed to be optimized for one-shot hash computation.

Reply:
Correction: `EVP_Digest()` is not available in 1.0.x, so I'll keep it as is.